temporal-sdk/src/main/java/io/temporal/internal/sync/AsyncInternal.java

```diff
@@ -280,8 +273,18 @@
     return WorkflowRetryerInternal.retryAsync(options, expiration, fn);
   }
 
-  private static <R> Promise<R> execute(boolean async, Functions.Func<R> func) {
-    if (async) {
+  /**
+   * Execute {@code func} asynchronously
+   *
+   * @param temporalStub If true, we work with temporal stub, it will be switched into async
+   *     invocation mode and promise of its call will be returned. If false, we work with a general
+   *     function and it will be triggered in a separate thread
+   * @param func function to call. Expected to be a lambda, method reference or {@code
+   *     Functions.Func} implementation
+   * @return promise on the return value of the asynchronous invocation of {@code func}
+   */
+  private static <R> Promise<R> execute(boolean temporalStub, Functions.Func<R> func) {
```

Comment:
Can `temporalStub` not be determined by inspecting `func` ?

Reply:
It can! but the point of this function is that we expected the `func` already earlier and provide this as an input. It's a private function, not a part of the class API.